Fleece/Array.hh

```diff
@@ -80,6 +80,7 @@ namespace fleece {
 
         friend class Value;
         friend class Dict;
+        friend class Arr;
```

Comment:
Actually Val and Arr aren't being used yet; they're some experimental/unfinished stuff that's checked in but shouldn't be built.